src/docs/sdk/hub_and_scope_refactoring.mdx

```diff
@@ -8,6 +8,38 @@ Development has started in JS and Python, but we are still ironing out the wrink
 
 All of Sentry's SDKs use the so-called <Link to="/sdk/unified-api/">Unified API</Link> under the hood. This API has some limitations and therefore we are refactoring it. This is a massive undertaking.
 
+## Why do we want to get rid of hubs?
+
+Hubs and scopes is a complicated system, that can easily be misused (also see [here](https://develop.sentry.dev/sdk/research/performance/#scope-propagation)). It requires us to create new hubs which in turn breaks breadcrumbs and other features. For the future we want to have one consistent system, that forks off in the same way as OTEL `Context` does.
+
+Regardless of switching our performance over to OTel, we should merge hubs and scopes. So hubs should be removed from SDKs.
+
+### Copy-on-write
+
+Getting rid of hubs also implies a soft copy-on-write.
+
+In our current system with hubs and scopes, `Scope` is always mutable and what makes it somewhat immutable is pushing/popping scopes and creating new hubs. This however has to be done manually.
+
+If an async operation is started halfway in between handling a request, a new hub clone has to be created to achieve copy-on-write. In case we don't create a new hub clone, popping a `Scope` also affects the original execution, leading to lost data or worst case exceptions.
+
+After merging hubs and scopes, scopes can be forked at any time and a user no longer has to take care of it, as it should automatically do the right thing. With this new system, the async execution can manipulate scope without affecting other execution.
+
+While it's still possible to misuse, this is a sort of soft copy-on-write that should protect most users.
```

Comment:
```suggestion
This soft copy-on-write relies on `Scope` being forked in certain cases:
- `withScope` and similar API wraps the callback in a forked `Scope` that'll be cleaned up by the SDK after the callback is finished
- we also have to fork, when OTel does, however there's some grey area here at the moment
	- it depends on what API is available in the OTel implementation relevant for each SDK
	- in some cases, we may only be able to intercept `Context` storage, i.e. when an already forked `Context` is being stored, e.g. in a ThreadLocal variable.
	- in some cases we can intercept `Context` forking, i.e. when a mutated copy of a `Context` is created
	- in some cases it might be necessary to have a mix of both, since not all auto instrumentation and user instrumentation go through a single place in the same way. e.g. in Java SDK we're sometimes handed a `Context` with some properties already set into our `ContextStorage` where we can then wrap it and intercept future forks of that `Context`
	- {open question} do we need to fork every time OTel `Context` copied or only when a new OTel span is set

While it's still possible to misuse, this is a sort of soft copy-on-write that should protect most users.
```